lua/telescope/config.lua

```diff
@@ -34,131 +58,273 @@ function config.set_pickers(pickers)
+      if xx == 0 and yy == 0 then
+        return ""
+      end
+
+      return string.format("%s / %s", xx, yy)
+    end,
+  },
+
+  dynamic_preview_title = {
+    false,
+    [[
+Will change the title of the preview window dynamically, where it
+is supported. Means the preview window will for example show the
+full filename.
+
+Default: false
+]],
+  },
 
   -- Builtin configuration
 
   -- List that will be executed.
   --    Last argument will be the search term (passed in during execution)
-  set("vimgrep_arguments",
-      {'rg', '--color=never', '--no-heading', '--with-filename', '--line-number', '--column', '--smart-case'}
-  )
-  set("use_less", true)
-  set("color_devicons", true)
-
-  set("set_env", nil)
-
-  -- TODO: Add motions to keybindings
-
-  -- To disable a keymap, put [map] = false
-  --        So, to not map "<C-n>", just put
-  --
-  --            ...,
-  --            ["<C-n>"] = false,
-  --            ...,
-  --
-  --        Into your config.
-  --
-  -- Otherwise, just set the mapping to the function that you want it to be.
-  --
-  --            ...,
-  --            ["<C-i>"] = actions.select_default
-  --            ...,
-  --
-  set("mappings", {})
-  set("default_mappings", nil)
-
-  set("generic_sorter", sorters.get_generic_fuzzy_sorter)
-  set("prefilter_sorter", sorters.prefilter)
-  set("file_sorter", sorters.get_fuzzy_file)
-
-  set("file_ignore_patterns", nil)
-
-  set("dynamic_preview_title", false, [[
-    Will change the title of the preview window dynamically, where it
-    is supported. Means the preview window will for example show the
-    full filename.
-
-    Default: false
-  ]])
-
-  set("file_previewer", function(...) return require('telescope.previewers').vim_buffer_cat.new(...) end)
-  set("grep_previewer", function(...) return require('telescope.previewers').vim_buffer_vimgrep.new(...) end)
-  set("qflist_previewer", function(...) return require('telescope.previewers').vim_buffer_qflist.new(...) end)
-  set("buffer_previewer_maker", function(...) return require('telescope.previewers').buffer_previewer_maker(...) end)
+  vimgrep_arguments = {
+    { "rg", "--color=never", "--no-heading", "--with-filename", "--line-number", "--column", "--smart-case" },
+  },
+
+  use_less = { true },
+
+  color_devicons = { true },
+
+  set_env = { nil },
+
+  mappings = {
+    {}, [[
+Your mappings to override telescope's default mappings.
+
+Format is:
+{
+  mode = { ..keys }
+}
+
+where {mode} is the one character letter for a mode ('i' for insert, 'n' for normal).
+
+For example:
+
+mappings = {
+  i = {
+    ["<esc>"] = actions.close,
+  },
+}
+
+
+To disable a keymap, put [map] = false
+      So, to not map "<C-n>", just put
+
+          ...,
+          ["<C-n>"] = false,
+          ...,
+
+      Into your config.
+
+
+otherwise, just set the mapping to the function that you want it to be.
+
+          ...,
+          ["<C-i>"] = actions.select_default
+          ...,
```

Comment:
This whole part is weirdly indented. I was also thinking maybe we should make this more complete.
Like in the builtin.init i put in the pickers configuration where i showed off more mappings:
```lua
      mappings = {
        i = {
          ["<c-d>"] = require("telescope.actions").delete_buffer,
          -- or right hand side can also be a the name of the action as string
          ["<c-d>"] = "delete_buffer",
        },
      }
```

Maybe we should also show the string option here and e.g.
```lua
["jj"] = { "<esc>", type = "command" },
["kk"] = { "<cmd>echo \"Hello, World!\"<cr>", type = "command" },
```